scripts/hudson/narayana.sh

```diff
@@ -123,11 +123,38 @@ function check_if_pull_closed
 function build_narayana {
   echo "Building Narayana"
   cd $WORKSPACE
-  [ $NARAYANA_TESTS = 1 ] && NARAYANA_ARGS= || NARAYANA_ARGS="-DskipTests"
 
-  ./build.sh -Prelease,community,all$OBJECT_STORE_PROFILE -Didlj-enabled=true "$@" $NARAYANA_ARGS $IPV6_OPTS -B clean install
+  if [ -n "$JAVA_HOME_FOR_BUILD" ]; then
+    local pathOriginal="$PATH"
+    local javaHomeOriginal="$JAVA_HOME"
+    export JAVA_HOME="$JAVA_HOME_FOR_BUILD"
+    export PATH="$JAVA_HOME/bin:$PATH"
+  fi
+
+  echo "Using Java version, Maven version with JAVA_HOME='$JAVA_HOME'"
+  ./build.sh -version
+
+  ./build.sh -Prelease,community,all$OBJECT_STORE_PROFILE -Didlj-enabled=true "$@" $NARAYANA_ARGS -Dtest=NonExistentTest -DfailIfNoTests=false $IPV6_OPTS -B clean install
   [ $? = 0 ] || fatal "narayana build failed"
 
+  if [ -n "$pathOriginal" ]; then
+     export PATH="$pathOriginal"
+     export JAVA_HOME="$javaHomeOriginal"
+  fi
+
+  return 0
+}
+
+function test_narayana {
```

Comment:
I wonder if we should put a commit like this on master too in case we need to do something similar in future? (ability to run tests with a different JDK)

Reply:
I note there is a further comment in https://github.com/jbosstm/narayana/pull/1730/commits/0323f19dec710bfd134f5bdd7536dca2d0b5bb33#diff-234fb7d762fdc80a3f29e8d8f4e0669ad47101d550eb3724420e91f64ee044de that could be useful to adapt too. But I expect some of the references to specific JDK versions like JDK6, 1.7 and 1.6 in the script and commit message would be revised perhaps to be more future looking.

Reply:
Currently there is not such need (as far as I know). If it sounds useful to you I can prepare it.
WDYT?